src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -156,6 +158,24 @@ public AddOnInfoAndVersions getByUid(String uid) throws IOException {
 		return client.execute(new Get.Builder(AddOnInfoAndVersions.ES_INDEX, uid).build())
 				.getSourceAsObject(AddOnInfoAndVersions.class);
 	}
+
+	@Override
+	public AddOnInfoAndVersions getByModulePackage(String modulePackage) throws IOException {
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(1)
+				.query(QueryBuilders.matchQuery("modulePackage", modulePackage)).toString())
```

Comment:
I think there should be a clause here to limit this to only type=OMOD. Please add this as a one-liner in a followup commit. (You don't need my review on that.)